### PR TITLE
Add UI functions for retrieving student information on section dates

### PR DESCRIPTION
**Pull request by DasilvaB**
added functions for:

- retrieving student courses
- retrieving a section id
- retrieving dates for a given section

Functions are listed at the bottom.

Again this is the last change where formatting will be updated.

### CODE REVIEW

**Review by jrm86 — APPROVED** [review]
Great work with this.  The formatting changes make the rest of it easier to read.

File: src/webapp/client/js/index.js [hunk 1]
```diff
@@ -357,3 +363,65 @@ function setAttendance(htmlText) {
...
 };
+
+// Returns list of courses for a given user id
+function getStudentCourses(connInfo, year, seasonorder, userrole) {
+	var urlParams = $.extend({}, connInfo, {year: year, seasonorder: seasonorder, userrole: userrole});
+	$.ajax('courses', {
+		dataType: 'json',
+		data: urlParams,
+		success: function(result) {
+			return result;
```
Comment (afig) [inline src/webapp/client/js/index.js]:
```suggestion
			//TODO: Implement population of calendar
```

File: src/webapp/client/js/index.js [hunk 2]
```diff
@@ -357,3 +363,65 @@ function setAttendance(htmlText) {
...
+		data: urlParams,
+		success: function(result) {
+			var sections = [];
+			for(var i = 0; i < result.sections.length; i++) {
+				sections.push({
+					"sectionid": result.sections[i].sectiondid,
+					"sectiontitle": result.sections[i].sectiontitle
+				})
+			}
+			return sections;
```
Comment (afig) [inline src/webapp/client/js/index.js]:
```suggestion
			//TODO: Implement population of calendar
```

File: src/webapp/client/js/index.js [hunk 3]
```diff
@@ -357,3 +363,65 @@ function setAttendance(htmlText) {
...
+	var urlParams = $.extend({}, connInfo, {sectionid: sectionid});
+	$.ajax('sectionsschedule', {
+		dataType: 'json',
+		data: urlParams,
+		success: function(result) {
+			var dates = [];
+			for(var i = 0; i < result.classDates.length; i++) {
+				sections.push(result.classDates[i]);
+			}
+			return dates;
```
Comment (afig) [inline src/webapp/client/js/index.js]:
```suggestion
			//TODO: Implement population of calendar
```

**Review by afig — COMMENTED** [review]
Its usually not possible to return values from an asynchronous call (which `.ajax` is). I recommend reading though the SO question and its top answer that I posted on MS Teams.

Essentially, whatever functionality was going to be implemented with the returned value instead has to be implemented here, withing the success portion of the call (ideally it would simply be a call to another function).

**Comment by DasilvaB** [comment]
Ooh, now i see why you made your calls like that. I'll take a look at how to re implement this.

**Review by afig — APPROVED** [review]
Looks good to merge. We can address the callback issues when the calendar is fully implemented

**Pull request merged**